x-pack/plugins/security_solution/public/resolver/store/data/selectors.ts

```diff
@@ -167,6 +168,105 @@ export function hasMoreAncestors(state: DataState): boolean {
   return tree ? resolverTreeModel.hasMoreAncestors(tree) : false;
 }
 
+/**
+ * A map of `entity_id`s to functions that provide information about
+ * related events by ECS `.category` Primarily to avoid having business logic
+ * in UI components.
+ */
+export const relatedEventInfoByEntityId = createSelector(
+  relatedEventsByEntityId,
+  relatedEventsStats,
+  function selectLineageLimitInfo(
+    /* eslint-disable no-shadow */
+    relatedEventsByEntityId,
+    relatedEventsStats
+    /* eslint-enable no-shadow */
+  ) {
+    const relatedEventInfoEntries = [...(relatedEventsStats || new Map()).entries()].map(
+      ([entityId, stats]) => {
+        const eventsResponseForThisEntry = relatedEventsByEntityId.get(entityId);
+        const hasMoreEvents =
+          eventsResponseForThisEntry && eventsResponseForThisEntry.nextEvent !== null;
+        /**
+         * Get the "aggregate" total for the event category (i.e. _all_ events that would qualify as being "in category")
+         * For a set like `[DNS,File][File,DNS][Registry]` The first and second events would contribute to the aggregate total for DNS being 2.
+         * This is currently aligned with how the backed provides this information.
+         *
+         * @param eventCategory {string} The ECS category like 'file','dns',etc.
+         */
+        const getAggregateTotalForCategory = (eventCategory: string): number => {
```

Comment:
can you change the name to `aggregateTotalForCategory` to stick w/ the variable naming convention in resolver

Reply:
https://github.com/elastic/kibana/pull/70097/commits/bc782255998493aca17dafaada196a05183bd511